pkg/proc/linutil/dynamic.go

```diff
@@ -0,0 +1,172 @@
+package linutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"github.com/go-delve/delve/pkg/proc"
+)
+
+const (
+	maxNumLibraries      = 1000000 // maximum number of loaded libraries, to avoid loading forever on corrupted memory
+	maxLibraryPathLength = 1000000 // maximum length for the path of a library, to avoid loading forever on corrupted memory
+)
+
+var ErrTooManyLibraries = errors.New("too many libraries")
+var ErrStringTooLong = errors.New("string too long")
```

Comment:
I think we can also make this more descriptive, unless there is context coming from a wrapping error.